dali/python/nvidia/dali/plugin/pytorch/__init__.pyi

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from typing import Union, Optional
+from typing import Any, Callable, Dict, List, Sequence
+
+import nvidia.dali.ops as ops
+from nvidia.dali import types as types
+from nvidia.dali.tensors import (
+    TensorCPU as TensorCPU,
+    TensorGPU as TensorGPU,
+    TensorListCPU as TensorListCPU,
+    TensorListGPU as TensorListGPU,
+)
+from nvidia.dali.data_node import DataNode
+from nvidia.dali.pipeline import Pipeline as Pipeline
+from nvidia.dali.plugin.base_iterator import LastBatchPolicy as LastBatchPolicy, _DaliBaseIterator
+
+def feed_ndarray(dali_tensor: Union[TensorCPU, TensorGPU, TensorListCPU, TensorListGPU],
+                 arr: torch.Tensor, cuda_stream: Union[torch.cuda.Stream, Any, None] = None):
+    ...
+
+
+class DALIGenericIterator(_DaliBaseIterator):
+
+    def __init__(
+        self,
+        pipelines: Union[List[Pipeline], Pipeline],
+        output_map: List[str],
+        size: int = -1,
+        reader_name: Optional[str] = None,
+        auto_reset: Union[str, bool, None] = False,
+        fill_last_batch: Optional[bool] = None,
+        dynamic_shape: Optional[bool] = ...,
+        last_batch_padded: bool = False,
+        last_batch_policy: LastBatchPolicy = LastBatchPolicy.FILL,
+        prepare_first_batch: bool = True,
+    ) -> None:
+        ...
+
+    def __next__(self) -> List[Dict[str, torch.Tensor]]:
+        ...
```

Comment:
I don't think that retyping everything here is the right thing to do - rather the contrary; it's prone to go stale in many subtle ways. Why can't we just annotate the proper code instead?

Reply:
The technical reason was to keep the TorchPythonFunction without changes, so I put the annotated signatures (that differs from the runtime ones) in the stub file, and thanks to that I need to put all the annotations in the stub, otherwise they become hidden (the annotation file has precedence over which symbols are visible, you can't mix and match this way).

I can move the TorchPythonFunction to a submodule, and move annotations for the iterators back to the file.

Reply:
Why do we need a submodule for that?

Reply:
So I can have TorchPythonFunction annotated with stub file, as this is needed to expand the definition of `**kwargs`.

Reply:
almost done